File: AspNetMvcUnleashed/0672329980_source/Chapter14/VB/MvcApplication1/MvcApplication1/Controllers/GuestBookController.vb
Public Class GuestBookController
    Inherits Controller

    Private _entities As New GuestBookDBEntities()

    ' GET: /GuestBook/

    Public Function Index() As ActionResult
        Return View(_entities.GuestSet.ToList())
    End Function

    ' POST: /GuestBook/Create

    Public Function Create(ByVal guestToCreate As Guest) As ActionResult
        _entities.AddToGuestSet(guestToCreate)
        _entities.SaveChanges()

        Return PartialView("Guests", _entities.GuestSet.ToList())
    End Function
End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter14/CS/MvcApplication1/MvcApplication1/Controllers/GuestBookController.cs
using System.Linq;
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class GuestBookController : Controller
    {
        private GuestBookDBEntities _entities = new GuestBookDBEntities();

        // GET: /GuestBook/

        public ActionResult Index()
        {
            return View(_entities.GuestSet.ToList());
        }

        // POST: /GuestBook/Create

        public ActionResult Create(Guest guestToCreate)
        {
            _entities.AddToGuestSet(guestToCreate);
            _entities.SaveChanges();

            return PartialView("Guests", _entities.GuestSet.ToList());
        }
    }
}
